oneview/data_source_interconnects.go

```diff
@@ -0,0 +1,1226 @@
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"consumed_count": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"license_type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"required_count": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"state": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"igmp_idle_timeout_interval": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"igmp_snooping_vlan_ids": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"initial_scope_uris": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"interconnect_ip": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interconnect_location": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"location_entries": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"interconnect_mac": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interconnect_type_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ip_address_list": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_address": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ip_address_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"lldp_ip_address_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"lldp_ipv4_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"lldp_ipv6_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"logical_interconnect_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"max_bandwidth": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"mgmt_interface": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"migration_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
```

Comment:
Name is required field as per  the docs/d/interconnects.html.markdown file above. So make it as required instead of optional.